src/cmdstan/command_helper.hpp

```diff
@@ -609,14 +609,40 @@ void services_log_prob_grad(const stan::model::model_base &model, bool jacobian,
  * this will throw an error.
  *
  * @param parser user config
+ * @param id id of initial chain or path. Only used for generated quantities
  * @return int num chains or paths
  */
-unsigned int get_num_chains(argument_parser &parser) {
+inline unsigned int get_num_chains(argument_parser &parser, unsigned int id) {
   auto user_method = parser.arg("method");
   if (user_method->arg("pathfinder"))
     return get_arg_val<int_argument>(parser, "method", "pathfinder",
                                      "num_paths");
 
+  auto gq_arg = user_method->arg("generate_quantities");
```

Comment:
Not crazy about this detection method - thoughts on gating this behavior behind a new `num_chains` argument similar to how multithreaded sampling works?

Reply:
Yeah I like that idea a lot more